### PR TITLE
[android] Fix saving to Google Drive

### PR DESCRIPTION
**Pull request by krozhdestvenski**
Fix issue https://github.com/organicmaps/organicmaps/issues/8806

### CODE REVIEW

File: android/app/src/main/java/app/organicmaps/util/SharingUtils.java [hunk 1]
```diff
@@ -234,6 +234,9 @@ public static void shareFile(Context context, ActivityResultLauncher<SharingInte
       // Properly set permissions for intent, see
       // https://developer.android.com/reference/androidx/core/content/FileProvider#include-the-permission-in-an-intent
       intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+
+      intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/util/SharingUtils.java:238]: Should it be inside `if (context instanceof Application)`, as it is done in other apps?

Comment (krozhdestvenski) [reply android/app/src/main/java/app/organicmaps/util/SharingUtils.java:238]: In this case context is Activity.

Comment (biodranik) [reply android/app/src/main/java/app/organicmaps/util/SharingUtils.java:238]: Does it allow to come back into Organic Maps after the file is exported?

Comment (krozhdestvenski) [reply android/app/src/main/java/app/organicmaps/util/SharingUtils.java:238]: Yes. After exporting we return to Organic Maps (checked it for exporting to Google Drive and local deevice)

Comment (biodranik) [reply android/app/src/main/java/app/organicmaps/util/SharingUtils.java:238]: There are several more places in the code where this function is called. Did you test it on different Android versions too? Does it work on Android 5?

Comment (krozhdestvenski) [reply android/app/src/main/java/app/organicmaps/util/SharingUtils.java:238]: i tested it in send bug report functionality.
It works correcctly on Android 10
On Android 5 send e-mail and savind to documents work too. Saving to Google Drive i can't check

Comment (biodranik) [reply android/app/src/main/java/app/organicmaps/util/SharingUtils.java:238]: Should FLAG_ACTIVITY_NEW_TASK be used when a single bookmark or POI is shared from the Place Page?

Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/util/SharingUtils.java:238]:
```suggestion
      intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION | Intent.FLAG_ACTIVITY_NEW_TASK);
```

Comment (krozhdestvenski) [reply android/app/src/main/java/app/organicmaps/util/SharingUtils.java]: On current version works correctly

**Review by Jean-BaptisteC — APPROVED** [review]
✅ Pixel 6 - Android 14
Tested with Telegram and GPX Viewer

**Review by biodranik — APPROVED** [review]
Thanks! Let's beta test it.

**Pull request merged**